src/core/model_accessing_API.jl

```diff
@@ -31,8 +31,7 @@ end
 
 function add_agent_to_model!(agent::AbstractAgent, model::StructABM)
     agent.id != nagents(model) + 1 && error(lazy"Cannot add agent of ID $(agent.id) in a vector ABM of $(nagents(model)) agents. Expected ID == $(nagents(model)+1).")
-    tuple_agent = NamedTuple(x => getfield(agent, x) for x in fieldnames(typeof(agent)))
-    push!(agent_container(model), tuple_agent)
+    push!(agent_container(model), agent)
```

Comment:
Rather, this should be a dedicated function `add_agent_to_container!`. And the function `add_agent_to_model!` shouldn't be named at all this way. We have `add_agent!`. This is the function that means "add agent to model", which entails more lower level processes: add to space, add to container, update next id, etc. etc. 

So all `add_agent!` functions propagate to the lowest `add_agent!(agent::AA, model::ABM)`. From there we call the internal functions `add_agent_to_container!`, to space, and the extra actions.

Reply:
yes, right, my change was from a performance standpoint